app/subscribers/super_good/solidus_taxjar/spree/reporting_subscriber.rb

```diff
@@ -3,12 +3,32 @@ module SolidusTaxjar
     module Spree
       module ReportingSubscriber
         include ::Spree::Event::Subscriber
+
         event_action :report_transaction, event_name: :shipment_shipped
+        event_action :replace_transaction, event_name: :order_recalculated
 
         def report_transaction(event)
           return unless SuperGood::SolidusTaxjar.reporting_enabled
+
           SuperGood::SolidusTaxjar::ReportTransactionJob.perform_later(event.payload[:shipment].order)
         end
+
+        def replace_transaction(event)
+          order = event.payload[:order]
+
+          return unless SuperGood::SolidusTaxjar.reporting_enabled
+
+          if order.complete? && order.paid? && amount_changed?(order)
+            SuperGood::SolidusTaxjar::ReplaceTransactionJob.perform_later(event.payload[:order])
+          end
+        end
+
+        private
+
+        def amount_changed?(order)
+          SuperGood::SolidusTaxjar.api.show_latest_transaction_for(order).amount !=
```

Comment:
We can't `Money.new` this because TaxJar doesn't provide a currency. We could assume USD, but I think it's weird to make that assumption.

This isn't a long lived object, it's just a float that is created by TaxJar and given to us... so maybe floating point precision isn't an issue?

Alex and I thought about this for a while and we feel kinda stuck... if this is critical to deal with now, perhaps we can discuss more on this PR or in person.

Reply:
It seems like solidus by default sets up the `spree_orders` table with fixed 2 decimals, and the taxjar API returns values with fixed two decimals.

However, `Taxjar::Order` stores the value on the model as a float, which definitely makes me still worry about floating point precision.... although you're right, maybe floating point precision isn't an issue.

How would you feel about using `BigDecimal` comparision with a truncation of decimals to `2` just to rule out floating point precision problems?

https://stackoverflow.com/a/4586130


![Screen Shot 2022-02-23 at 13 34 30PST](https://user-images.githubusercontent.com/8933450/155412407-6bf1c3c4-5f8a-4867-8a57-c1ef4cee4f21.png)


I'm also down to just drop this for this PR, the tests are passing and this probably is a non-issue. Float comparisons just scare me ahahah

Reply:
I captured this as a new issue: #167